### PR TITLE
fix: Add file extensions to all files

### LINKED ISSUE #684: File extensions
Hi everyone :wave:
Thanks for this awesome Project!

Pelorus have some files without extension. To get the list of files without extension, in the project root run 
```
find . -type f ! -name "?*.*" | grep -v "venv/\|git/"
```
Some of them should not have extension (like `.gitignore`), but I believe the Python files should, for example. This leads to the project quality checks not checking files properly.

For example, running `make black-check` does not warn of any errors today. But if we change the name of the file `scripts/bump-version` to `scripts/bump-version.py` and run `make black-check` again, black indicates format errors.

### PR DESCRIPTION
**Pull request by mateusoliveira43**
Signed-off-by: Mateus Oliveira <msouzaol@redhat.com>

## Describe the behavior changes introduced in this PR

Add file extensions to all files. This enable the quality tools to check all of them.

We can add them to makefile or console scripts to avoid calling them with the extension.

## Linked Issues

resolves #684
depends on #944 

## Testing Instructions

I have run `find . -type f ! -name "?*.*" | grep -v "venv/\|git/\|.pytest_cache/\|htmlcov/"` to check for files without extension (there is still one that I think should have, that is `./pelorus-operator/PROJECT`, but it is generated by script, so we need to change script to change it).

To check if I have changed all needed places of invocation of the file, I have run `grep -Iinr 'name_of_the_file' . --exclude-dir={.venv,htmlcov,.pytest_cache,.mypy_cache}`.

### CODE REVIEW

File: scripts/chart-lint.py [hunk 1]
```diff
@@ -10,3 +10,5 @@
     parser.parse_args()
     exit, _ = run_chart_lint()
     sys.exit(exit)
+
+# it is not used by CI, how to use it? what it does that make chart-lint does not?
```
Comment (mateusoliveira43) [inline scripts/chart-lint.py:14]: I did not see any mention to this file

File: scripts/create_pelorus_operator.sh [hunk 2]
```diff
@@ -179,3 +179,6 @@ echo "  # # Deploy bundle, e.g. version v${NEW_OPERATOR_VERSION} in the pelorus
 echo "  # operator-sdk run bundle quay.io/${OPERATOR_ORG_NAME}/pelorus-operator-bundle:v${NEW_OPERATOR_VERSION} --namespace pelorus"
 echo "  # # Create pelorus instance from example in the pelorus namespace"
 echo "  # oc apply -f pelorus-operator/config/samples/charts_v1alpha1_pelorus.yaml -n pelorus"
+
+
+# Not documented!
```
Comment (mateusoliveira43) [inline scripts/create_pelorus_operator.sh:184]: I did not see any mention to this file

File: Makefile [hunk 3]
```diff
@@ -239,7 +239,7 @@ endif
 
 shellcheck:
 	@echo "🐚 📋 Linting shell scripts with shellcheck"
-	$(SHELLCHECK) $(find -name '*.sh' -type f | grep -v "venv/\|git/\|.pytest_cache/\|htmlcov/")
+	$(SHELLCHECK) $(shell find -name '*.sh' -type f | grep -v 'venv/\|git/\|.pytest_cache/\|htmlcov/')
```
Comment (mpryc) [inline Makefile:242]: How about searching for all shell files (we can add exclude list similarly to .venv example):
```shell
$ find . -type f ! -path "*.venv/*" | xargs file -i - | grep "text/x-shellscript" | awk -F":" '{print $1}'
```

Comment (mateusoliveira43) [reply Makefile:242]: I tested and this command find less files than the one I am using (it did not find `_test/conftest.sh`)

Comment (mpryc) [reply Makefile:242]: That is correct. The `_test/conftest.sh` is not an shell file. Look at the first line:
```
#!/usr/bin/env bats
```

So it's bats interpreted file not shell and as such shellcheck should not be ran against it.

Comment (mateusoliveira43) [reply Makefile:242]: I think that is not a problem https://github.com/koalaman/shellcheck/issues/417

or do you want to only check bash/sh files?

File: _test/bats-support-clone.sh [hunk 4]
```diff
@@ -1,9 +1,10 @@
+#!/bin/bash
```
Comment (mateusoliveira43) [inline _test/bats-support-clone.sh:1]: conftest failed with this change, study how to do it

File: Makefile [hunk 5]
```diff
@@ -262,3 +254,8 @@ endif
 clean-dev-env:
 	rm -rf ${PELORUS_VENV}
 	find . -iname "*.pyc" -delete
+
+PYMA=$(shell which python)
```
Comment (mateusoliveira43) [inline Makefile:258]: remove, this was a test

File: Makefile (hunk 3 above)
Comment (mpryc) [reply Makefile:242]: I would be more careful and check only bash/sh files. There can be some list of *additional* files to be checked if they are not proper  `"text/x-shellscript"` type.

File: _test/bats-support-clone.sh (hunk 4 above)
Comment (mpryc) [reply _test/bats-support-clone.sh:1]: See above. I would really ad this file to some exception list to be checked, or forget to shellscript this one all together.

File: scripts/chart-lint.py (hunk 1 above)
Comment (mateusoliveira43) [reply scripts/chart-lint.py]: deleted

File: Makefile (hunk 5 above)
Comment (mateusoliveira43) [reply Makefile:258]: done

File: _test/bats-support-clone.sh (hunk 4 above)
Comment (mateusoliveira43) [reply _test/bats-support-clone.sh]: alright, remove this one from shellcheck

File: Makefile (hunk 3 above)
Comment (mateusoliveira43) [reply Makefile]: I remove the bats file from the shellcheck with grep (to have a simpler command), but added a note here https://github.com/konveyor/pelorus/issues/825 to fix it later

File: scripts/create_pelorus_operator.sh (hunk 2 above)
Comment (mateusoliveira43) [reply scripts/create_pelorus_operator.sh:188]: opened a issue to fix this https://github.com/konveyor/pelorus/issues/841

**Comment by mateusoliveira43** [comment]
can we delete `demo/deprecrated/demo.sh` file?

File: .github/workflows/shellcheck.yml [hunk 6]
```diff
@@ -1,26 +1,11 @@
...
     paths:
-      - 'demo/demo-tekton.sh'
-      - 'scripts/create_release_pr'
-      - 'scripts/install_dev_tools'
-      - 'scripts/pre-commit'
-      - 'scripts/run-mockoon-tests'
-      - 'scripts/run-pelorus-e2e-tests'
-      - 'scripts/setup-pre-commit-hook'
-
+      - '**.sh'
```
Comment (mateusoliveira43) [inline .github/workflows/shellcheck.yml:5]: check the other workflows files

Comment (mateusoliveira43) [reply .github/workflows/shellcheck.yml:7]: done

File: .github/workflows/shellcheck.yml [hunk 7]
```diff
@@ -1,26 +1,15 @@
...
-      - 'scripts/create_release_pr'
-      - 'scripts/install_dev_tools'
-      - 'scripts/pre-commit'
-      - 'scripts/run-mockoon-tests'
-      - 'scripts/run-pelorus-e2e-tests'
-      - 'scripts/setup-pre-commit-hook'
-
+      - '.github/workflows/shellcheck.yml'
+      - 'Makefile'
+      - '**.sh'
```
Comment (mpryc) [inline .github/workflows/shellcheck.yml:7]: Should this be?:

      - '**/*.sh'

File: .github/workflows/python-linting.yml [hunk 8]
```diff
@@ -2,11 +2,10 @@ name: Pylava
 on:
   pull_request:
     paths:
-      - 'exporters/**'
-      - 'scripts/**'
       - 'pyproject.toml'
       - '.github/workflows/python-linting.yml'
       - 'Makefile'
+      - '**.py'
```
Comment (mpryc) [inline .github/workflows/python-linting.yml:8]: Similar to .sh, should it be?

      - '**/*.py'

File: .github/workflows/python-formatting.yml [hunk 9]
```diff
@@ -2,10 +2,10 @@ name: Check Python Formatting
 on:
   pull_request:
     paths:
-      - 'exporters/**'
-      - 'pyproject.toml'
       - '.github/workflows/python-formatting.yml'
+      - 'pyproject.toml'
       - 'Makefile'
+      - '**.py'
```
Comment (mpryc) [inline .github/workflows/python-formatting.yml:8]: Same as others, cause it won't catch all subdirs:

      - '**/*.py'

**Comment by mpryc** [comment]
> can we delete `demo/deprecrated/demo.sh` file?

Yes, but another PR, so we don't hide such removals within other changes (easier to restore or track them if needed)

**Review by mpryc — APPROVED** [review]

**Review by mpryc — CHANGES_REQUESTED** [review]
Fix shell or disable broken file in this PR and fix in following PR.

**Review by mpryc — DISMISSED** [review]

**Pull request merged**